### PR TITLE
Rework collection references as separate entities

### PR DESCRIPTION
**Pull request by dstockwell**
- [x] move `CollectionReference` from `Library` to standalone entity
- [x] filter collection list in metadata to only one entry per collection
- [x] write tests for:
 - [x] collection ingestion
 - [x] deleting stale collection references

### CODE REVIEW

**Comment by dstockwell** [comment]
Still TODO:
- [ ] filter collection list in metadata to only one entry per collection
- [ ] write tests for:
 - [ ] collection ingestion
 - [ ] deleting stale collection references

File: src/datamodel.py [hunk 1]
```diff
@@ -99,12 +102,31 @@ class Version(ndb.Model):
...
+      if collection_version is None:
+        # Remove the stale reference.
+        yield collection_references[i].key.delete_async()
+      elif versiontag.match(version_key.id(), collection_references[i].semver):
+        collection_id = collection_version.key.parent().id()
+        existing_version = result_map.get(collection_id, None)
+        if existing_version is None or versiontag.compare(collection_version.key.id(), existing_version.key.id()) > 0:
+          result_map[collection_id] = collection_version
+      else:
+        raise Exception([version_key.id(), collection_references[i].semver])
```
Comment (shans) [inline src/datamodel.py]: ?

**Comment by shans** [comment]
lgtm

File: src/datamodel.py (hunk 1 above)
Comment (dstockwell) [inline src/datamodel.py]: > ?

Oops. Removed and added test.

**Pull request merged**